### DISCUSSION TITLE
Free Code

### DISCUSSION THREAD

**Discussion by anxofcostas** [NetSparkleUpdater/NetSparkle#378] [Q&A]
Hello! I like so much your project. I need to modificate some things of your project to adapt to my idea. Can you send my the code without lock to change this aspects? Thank you!!

**Comment by Deadpikle** [comment]
Hi @anxofcostas, glad you're enjoying things and that it's useful. The code is already here on this repository, [MIT licensed](https://tldrlegal.com/license/mit-license), and you can easily fork the code and modify it or just download it for your own needs. Have fun!

**Reply by anxofcostas** [reply]
Hello @Deadpikle !
I already dowloaded the code but some Classes i can´t modify them. What Can i do?

**Reply by Deadpikle** [reply]
What do you mean? If you downloaded the code, you have full and complete control over literally every aspect of the project other than the license, which is open such that you can do what's needed. Just change what you need to change. What exactly do you mean?

**Reply by anxofcostas** [reply]
I'm sorry. Now I already know how to do it.
Another question. When I will finish to modify it, how can export my new NetSparkle like a library?

**Reply by Deadpikle** [reply]
It's OK. :) Just build the project (using Visual Studio or `dotnet build` on the command line) and it will output a .DLL file that you can reference in whatever project you are working on.

**Reply by anxofcostas** [reply]
And inside which folder can I find that .DLL ?
Thank you so much!

**Reply by anxofcostas** [reply]
Other question is: When I install and open the project and I don´t modify anything, I get a Error: Severity Code Description Project Project File Line Status suppressed
Error Could not restore NuGet package for NetSparkle.Xamarin project: 'NuGet.Build.Packaging' package version '0.2.5-dev.8' could not be found.
  C:\Program Files (x86)\Microsoft SDKs\NuGetPackages\: The package 'NuGet.Build.Packaging.0.2.5-dev.8' was not found in the source 'C:\Program Files (x86)\Microsoft SDKs\NuGetPackages\'.
  https://www.myget.org/F/avalonia-ci/api/v2: 'NuGet.Build.Packaging.0.2.5-dev.8' package was not found in source 'https://www.myget.org/F/avalonia-ci/api/v2'.
  https://api.nuget.org/v3/index.json: Package 'NuGet.Build.Packaging.0.2.5.5-dev.8' was not found in source 'https://api.nuget.org/v3/index.json'.
. See the Error List window for more detailed errors and warnings.				

How can I solve it?

**Reply by anxofcostas** [reply]
> And inside which folder can I find that .DLL ? Thank you so much!

Because I can reference NetSparkle dll, but i want to reference the wholw project

**Reply by Deadpikle** [reply]
> And inside which folder can I find that .DLL ?

When you build the NetSparkle library (or one of the UI ones) inside of Visual Studio, it will show you the output location in the Output window. See also https://learn.microsoft.com/en-us/visualstudio/ide/reference/output-window?view=vs-2022. Generally speaking, for NetSparkle, this is in a subfolder of `[Path to NetSparkle folder]/src/bin`. You'll also see the output folder via the `dotnet build` action.

> When I install and open the project and I don´t modify anything, I get a Error: Severity Code Description Project Project File Line Status suppressed

The `NetSparkle.Xamarin` project is not complete, nor is it functional on Windows. Don't worry about this or try to build it that project.

> Because I can reference NetSparkle dll, but i want to reference the wholw project

If you want to reference the whole project, it depends on your setup as to how you'll want to do this. Generally speaking you can copy the code to your code folder and add the NetSparkle project(s) to your solution file (`.sln`). See also https://stackoverflow.com/questions/47634380/adding-reference-of-another-external-project-in-a-solution. But it depends on your setup. See also [https://learn.microsoft.com/en-us/dotnet/core/tools/dotnet-sln](`dotnet sln add`).
